src/core/import-creds.c

```diff
@@ -277,6 +277,96 @@ static int import_credentials_boot(void) {
         return 0;
 }
 
+static int import_credentials_boot_menu(void) {
+        _cleanup_(import_credentials_context_free) ImportCredentialContext context = {
+                .target_dir_fd = -EBADF,
+        };
+        _cleanup_free_ DirectoryEntries *de = NULL;
+        _cleanup_close_ int source_dir_fd = -EBADF;
+        int r;
+
+        /* systemd-stub will wrap credentials typed via the boot menu as files for the initrd */
+
+        if (!in_initrd())
+                return 0;
+
+        source_dir_fd = open("/.extra/synthetic_credentials/", O_RDONLY|O_DIRECTORY|O_CLOEXEC|O_NOFOLLOW);
+        if (source_dir_fd < 0) {
+                if (errno == ENOENT) {
+                        log_debug("No credentials passed via /.extra/synthetic_credentials/.");
+                        return 0;
+                }
+
+                return log_warning_errno(errno, "Failed to open '/.extra/synthetic_credentials/': %m");
+        }
+
+        r = readdir_all(source_dir_fd, RECURSE_DIR_SORT|RECURSE_DIR_IGNORE_DOT, &de);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to read '/.extra/synthetic_credentials/' contents: %m");
+
+        for (size_t i = 0; i < de->n_entries; i++) {
+                const struct dirent *d = de->entries[i];
+                _cleanup_close_ int cfd = -EBADF, nfd = -EBADF;
+                struct stat st;
+
+                if (!credential_name_valid(d->d_name)) {
+                        log_warning("Credential '%s' has invalid name, ignoring.", d->d_name);
+                        continue;
+                }
+
+                cfd = openat(source_dir_fd, d->d_name, O_RDONLY|O_CLOEXEC);
+                if (cfd < 0) {
+                        log_warning_errno(errno, "Failed to open %s, ignoring: %m", d->d_name);
+                        continue;
+                }
+
+                if (fstat(cfd, &st) < 0) {
+                        log_warning_errno(errno, "Failed to stat %s, ignoring: %m", d->d_name);
+                        continue;
+                }
+
+                r = stat_verify_regular(&st);
+                if (r < 0) {
+                        log_warning_errno(r, "Credential file %s is not a regular file, ignoring: %m", d->d_name);
+                        continue;
+                }
+
+                if (!credential_size_ok(&context, d->d_name, st.st_size))
+                        continue;
+
+                r = acquire_credential_directory(&context, SYSTEM_CREDENTIALS_DIRECTORY, /* with_mount= */ false);
```

Comment:
uh? i am sorry, but NO!

you cannot just write unauthenticated creds into the dir for trusted, authenticated creds. That's not how this works. We establish trust first, and only then allow the data through the flood gates.

How do you intend to establish trust into creds the user inputs?

Reply:
From the PR description:

> Initial draft, not yet ready. Menu prompt is very crude. But enough to get started: **do we mark these as untrusted? Merge with unencrypted? Or add a third category with yet another unit option?** Do we measure them?